lib/akismet.js

```diff
@@ -73,17 +75,15 @@ var Akismet = function(options) {
       referrer             : options.referrer || options.referer || ''
     })
     .set('User-Agent', this.userAgent)
-    .end(function(err, res) {
-      if (err) cb(err, null);
-      else if (res.text == 'true') cb(null, true);
-      else if (res.text == 'false') cb(null, false);
-      else if (res.header['x-akismet-debug-help']) 
-        cb(new Error(res.header['x-akismet-debug-help']), null);
-      else if (res.text == 'invalid')
-        cb(new Error('Invalid API key'), null);
-      else 
-        cb(new Error(res.text), null);
-    });
+    .endAsync()
+    .then(function(res) {
+      if (res.text == 'true') return true;
```

Comment:
Is 'res' or 'res.text' always set? Also, there is no 'err' parameter anymore, is that intentional? Same question for the function above.